golang/vaas/v2/pkg/vaas/vaas.go

```diff
@@ -79,29 +129,21 @@ func NewWithDefaultEndpoint(options options.VaasOptions, authenticator authentic
 func (v *vaas) ForSha256(ctx context.Context, sha256 string) (msg.VaasVerdict, error) {
 	// Loop until we get 200 or an error
 	for {
-		token, err := v.authenticator.GetToken()
-		if err != nil {
-			return msg.VaasVerdict{}, err
+		select {
+		case <-ctx.Done():
```

Comment:
Do we need this?

Reply:
Technically no, but it's generally a bit more user friendly (prevents triggering a HTTP request when the ctx is already cancelled). If we are fine with creating a garbage request (that is immediately going to be thrown away during I/O), we can remove this.